src/bot/handlers.py

```diff
@@ -58,10 +58,16 @@ async def categories_callback(update: Update, context: ContextTypes.DEFAULT_TYPE
 
 async def subcategories_callback(update: Update, context: ContextTypes.DEFAULT_TYPE):
     query = update.callback_query
-    parent_id = int(query.data.split("_")[1])
+    parent_id_match = context.match
+
+    if not parent_id_match:

```

Comment:
Зачем это определять в отдельную переменную? Можно проще 
```suggestion
    if not context.match:
```

К тому же, если не будет context.match хендлер не запуститься - поэтому эта проверка не имеет смысла. Если посмотришь исходники, там вообще в случае если нет match-а атрибут не задается (обращение к `context.match` - будет падать с ошибкой AttributeError), но это может произойти только если паттерн не указан.